daemon/graphdriver/zfs/zfs.go

```diff
@@ -357,37 +358,45 @@ func (d *Driver) Remove(id string) error {
 }
 
 // Get returns the mountpoint for the given id after creating the target directories if necessary.
-func (d *Driver) Get(id, mountLabel string) (containerfs.ContainerFS, error) {
+func (d *Driver) Get(id, mountLabel string) (_ containerfs.ContainerFS, retErr error) {
 	mountpoint := d.mountPath(id)
 	if count := d.ctr.Increment(mountpoint); count > 1 {
 		return containerfs.NewLocalContainerFS(mountpoint), nil
 	}
+	defer func() {
+		if retErr != nil {
+			if c := d.ctr.Decrement(mountpoint); c <= 0 {
```

Comment:
Are there any locks here to prevent another process from increment at the same time this decrements?

Reply:
Yes, `d.ctr` is of `graphdriver.RefCounter` type which does inc/dec under a mutex (see https://github.com/moby/moby/blob/master/daemon/graphdriver/counter.go)